### PR TITLE
User feed

### PR DESCRIPTION
**Pull request by chanchal16**
- Created user feed page
- list all posts
- create a post
- delete a post

https://user-images.githubusercontent.com/56486295/168487318-b1053194-c43e-45b8-a9ee-5e4b63a564be.mp4

### CODE REVIEW

File: src/features/posts/postSlice.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      .addCase(createPost.rejected, (state) =>{
+        state.status = 'Rejected';
+      })
+      // delete post
+			.addCase(deletePost.fulfilled, (state, action) => {
+				state.allPosts = state.allPosts.filter(post=> post._id !== action.meta.arg)
+				state.userPosts = state.userPosts.filter(post=>post._id !== action.payload)
+			})
+    }
+  })
```
Comment (bharati-21) [inline src/features/posts/postSlice.js:89]: Great to see that you have followed the docs convention to handle actions in extraReducers!

File: src/pages/Feed.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    useEffect(() => {
+        dispatch(getPosts())
+    }, [dispatch])
+    
+  return (
+    <div className='grid grid-cols-6 gap-x-4 lg:gap-x-8 w-full my-4'>
+
+        <div className="col-span-6 md:col-span-4">
+            {/* <CreatePost/>  */}
```
Comment (bharati-21) [inline src/pages/Feed.jsx:20]: Please remove commented code.

File: src/pages/Login.jsx [hunk 3]
```diff
@@ -35,6 +36,7 @@ export const Login = () => {
             email: "testing@test.com",
```
Comment (bharati-21) [inline src/pages/Login.jsx:36]: Suggestion- it's a good practice to use keys/ credentials using .env file.

File: src/store/store.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 import { configureStore } from '@reduxjs/toolkit'
-import authReducer from 'features/authentication/authSlice'
+import authReducer from 'features/authentication/authSlice';
+import postReducer from 'features/posts/postSlice'
```
Comment (bharati-21) [inline src/store/store.js:3]: Suggestion-You can try barrel exports and export from features/ Will be helpful as your slice code grows.

**Review by bharati-21 — APPROVED** [review]
Easy to read code.
My only suggestion is that you could use a formatter as your tabs, line breaks and spacings are quiet inconsistent.

Otherwise, looks good to be merged!

**Review by sandeepkumardev — APPROVED** [review]

File: src/store/store.js (hunk 4 above)
Comment (chanchal16) [reply src/store/store.js:3]: ya but they are in different folders

File: src/pages/Login.jsx (hunk 3 above)
Comment (chanchal16) [reply src/pages/Login.jsx:36]: ok, will look into it.

**Comment by chanchal16** [comment]
> Easy to read code. My only suggestion is that you could use a formatter as your tabs, line breaks and spacings are quiet inconsistent.
> 
> Otherwise, looks good to be merged!

Thanks for the suggestions!

**Pull request merged**